src/engine/strat_engine/filesystem.rs

```diff
@@ -1,17 +1,42 @@
 // This Source Code Form is subject to the terms of the Mozilla Public
 // License, v. 2.0. If a copy of the MPL was not distributed with this
 // file, You can obtain one at http://mozilla.org/MPL/2.0/.
+extern crate rand;
 
 use uuid::Uuid;
+use devicemapper::DM;
+use engine::EngineResult;
 
 use engine::Filesystem;
-
+use engine::strat_engine::thindev::ThinDev;
+use engine::strat_engine::thinpooldev::ThinPoolDev;
 use super::super::engine::{HasName, HasUuid};
+use types::Sectors;
 
 #[derive(Debug)]
 pub struct StratFilesystem {
     fs_id: Uuid,
     name: String,
+    thin_dev: ThinDev,
+}
+
+impl StratFilesystem {
+    pub fn new(fs_id: Uuid,
+               name: &str,
+               dm: &DM,
+               thin_pool: &mut ThinPoolDev)
+               -> EngineResult<StratFilesystem> {
+        // TODO should replace with proper id generation.
+        let thin_id = rand::random::<u16>();
```

Comment:
I see, it needs to be passed to the dm-table. Can you make line above:

```let thin_id = rand::random::<u32>();``` to avoid the widening cast below? It _ought_ to be more random that way.

If devicemapper really requires u32s, then I guess I don't see anything particularly wrong with just generating random ones. What else should we do?